### PR TITLE
Write lambda function get s3 object

### PR DESCRIPTION
**Pull request by Chufle**
finished lambda function for photo-upload-workflow. function works and does:
- detects new file on s3 bucket "nfish-des-kutter-photos"
- writes metadata to dynamodb table "kutter-table"

change terraform files to make it work with:
- iam role with attached policies for s3, lambda, logs and dynamodb (policies allows too much, change soon)
- s3 bucket "nfish-des-kutter-photos" created in file "lambda_photo_checkin.tf" instead of shell-script

### CODE REVIEW

File: infrastructure/iam_for_lambda.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "nfish-des-lambda_s3_access" {
+  role       = aws_iam_role.nfish-des-role-lambda_s3_access.name
+  for_each = toset([
+  "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+  "arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess",
```
Comment (fabianschmauder) [inline infrastructure/iam_for_lambda.tf:25]: To assign full access to a resource is bad practice

Try to specify the actions and resources.
Is also a good practice

File: photo-checkin/src/photo_checkin.py [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
-        print("originalFileName: " + key)
-        print("s3Bucket: " + bucket)
-        print("creationDate: " + event['Records'][0]['eventTime'])
-    except Exception as e:
-        print(e)
-        print('Error getting object {} from bucket {}. Make sure they exist and your bucket is in the same region as this function.'.format(key, bucket))
-        raise e
+def put_db_object(object_id, file_name, bucket, creation_date):
+    dynamodb = boto3.resource('dynamodb')
+    table = dynamodb.Table('kutter-table')
```
Comment (fabianschmauder) [inline photo-checkin/src/photo_checkin.py:17]: you could get the table name from env

File: photo-checkin/src/photo_checkin.py [hunk 3]
```diff
@@ -1,20 +1,29 @@
 import urllib.parse
 import boto3
+import uuid
 
-print('Loading function')
+def get_s3_object(event):
+    bucket = event['Records'][0]['s3']['bucket']['name']
```
Comment (fabianschmauder) [inline photo-checkin/src/photo_checkin.py:6]: would be better to use a loop over the records and handle every record

File: photo-checkin/src/photo_checkin.py [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
-        response = s3.get_object(Bucket=bucket, Key=key)
-        print("originalFileName: " + key)
-        print("s3Bucket: " + bucket)
-        print("creationDate: " + event['Records'][0]['eventTime'])
-    except Exception as e:
-        print(e)
-        print('Error getting object {} from bucket {}. Make sure they exist and your bucket is in the same region as this function.'.format(key, bucket))
-        raise e
+def put_db_object(object_id, file_name, bucket, creation_date):
+    dynamodb = boto3.resource('dynamodb')
```
Comment (fabianschmauder) [inline photo-checkin/src/photo_checkin.py:16]: I would initialize the resource outside of the function

**Review by fabianschmauder — COMMENTED** [review]
Just a few small things :)

File: infrastructure/iam_for_lambda.tf (hunk 1 above)
Comment (Chufle) [reply infrastructure/iam_for_lambda.tf:25]: ok

File: photo-checkin/src/photo_checkin.py (hunk 4 above)
Comment (Chufle) [reply photo-checkin/src/photo_checkin.py:16]: ok. done in next pull request

File: photo-checkin/src/photo_checkin.py (hunk 2 above)
Comment (Chufle) [reply photo-checkin/src/photo_checkin.py:17]: done in next pull request

**Pull request merged**